Use newer VS UX for data source tests

Evangelink opened the pull request:
## Description

Using the newer version of the test fx + adding the private ctor allows us to have children of a test for each configuration rather than having multiple outputs.

Evangelink committed:
Use newer VS UX for data source tests

Evangelink commented on scripts/build/TestPlatform.Dependencies.props:
```diff
@@ -13,8 +13,8 @@
-    <MSTestFrameworkVersion>2.1.0</MSTestFrameworkVersion>
-    <MSTestAdapterVersion>2.1.0</MSTestAdapterVersion>
+    <MSTestFrameworkVersion>2.2.8</MSTestFrameworkVersion>
+    <MSTestAdapterVersion>$(MSTestFrameworkVersion)</MSTestAdapterVersion>
```
This is so that we have both versions in sync because I can't really think of a case where we would want different versions.

Evangelink replied:
This can't be done, integration tests are reading the file.

Evangelink committed:
Revert variable reference

Haplois commented on test/Microsoft.TestPlatform.AcceptanceTests/Extension/RunnnerInfo.cs:
```diff
@@ -9,7 +9,13 @@ namespace Microsoft.TestPlatform.AcceptanceTests;
-    public RunnerInfo(string runnerType, string targetFramework) : this(runnerType, targetFramework, "")
+    // Serialization ctor that enables correct tree view for data source tests
+    private RunnerInfo()
+    {
```
`IsNetRunner`, `IsNetFrameworkRunner`, `IsNetTarget`, `IsNetFrameworkTarget` won't be correctly set if this gets deserialized. We probably should switch to Lazy<bool> and calculate those values on first fetch.

Evangelink committed:
Use record for RunnerInfo

Evangelink commented:
@Haplois I have reworked the class to be a record (I also did some tests to check property values are preserved).

nohwnd commented on test/Microsoft.TestPlatform.AcceptanceTests/Extension/RunnnerInfo.cs:
```diff
@@ -1,84 +1,35 @@
-
-    public override string ToString()
-    {
-        return string.Join(",", new[] { "RunnerFramework = " + RunnerFramework, " TargetFramework = " + TargetFramework, string.IsNullOrEmpty(InIsolationValue) ? " InProcess" : " InIsolation" });
```
Where did this go?

nohwnd reviewed (changes requested):
Put the serialization back, it is used to print the data in test output.

nohwnd commented on test/Microsoft.TestPlatform.AcceptanceTests/Extension/RunnnerInfo.cs:
```diff
@@ -1,84 +1,35 @@
 
     /// <summary>
-    /// Is running via .NET "Core" testhost?
+    /// Is running via .NET Framework testhost?
```
```suggestion
    /// Is running via .NET "Core" testhost?
```

nohwnd commented on test/Microsoft.TestPlatform.AcceptanceTests/Extension/RunnnerInfo.cs:
```diff
@@ -1,84 +1,35 @@
     /// </summary>
-    /// <returns></returns>
-    public bool IsNetRunner { get; }
+    public bool IsNetRunner => RunnerFramework.StartsWith("netcoreapp", StringComparison.InvariantCultureIgnoreCase);
```
if you are making changes maybe change this to be on the IsNetFramework runner and say StartsWith "net4"?

Evangelink commented on test/Microsoft.TestPlatform.AcceptanceTests/IsExternalInit.cs:
```diff
@@ -0,0 +1,20 @@
+/// /!\ Do not use directly. /!\
+/// </summary>
+[ExcludeFromCodeCoverage, DebuggerNonUserCode]
+internal static class IsExternalInit
```
Ideally we would want this file to be added to all projects but keeping that for later.

Evangelink replied:
Ideally we would want this file to be added to all projects but keeping that for later.

Evangelink committed:
Address review comments

Evangelink committed:
Fix expected test name

In the new version we now have a display name so we need to update the expected

Evangelink commented on test/Microsoft.TestPlatform.AcceptanceTests/LoggerTests.cs:
```diff
@@ -192,7 +192,7 @@ private void IsFileAndContentEqual(string filePath)
 
         string filePathContent = sb.ToString();
-        string[] divs = { "Total tests", "Passed", "Failed", "Skipped", "Run duration", "Pass percentage", "SampleUnitTestProject.UnitTest1.PassingTest" };
+        string[] divs = { "Total tests", "Passed", "Failed", "Skipped", "Run duration", "Pass percentage", "PassingTest" };
```
Test method now has a display name so the html logger no longer writes the fully qualified name.

nohwnd commented on test/Microsoft.TestPlatform.AcceptanceTests/Extension/RunnnerInfo.cs:
```diff
@@ -16,20 +16,30 @@ public record RunnerInfo(string RunnerFramework, string TargetFramework, string
+                "RunnerFramework = " + RunnerFramework,
+                " TargetFramework = " + TargetFramework,
+                string.IsNullOrEmpty(InIsolationValue) ? " InProcess" : " InIsolation",
+            });
```
I am not going to force you to build this again. But with all your comments on using interpolation, I expected that this will be gone as well 😛
